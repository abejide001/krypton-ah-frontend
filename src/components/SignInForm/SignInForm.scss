input#_password,input#password{
  width: 20rem !important;
  margin-left: 0.4rem;
}

.ui.form {
a {
    top: 55% !important;
    font-size: 0.7rem !important;
    line-height: 15px !important;
    background: rgba(226, 226, 226, 0.33) !important;
}
}

.ui.form .fields .field .ui.input input {
    width: auto;
    border: 1px solid #22242626;
}

input#_password,input#password {
  width: 20rem !important;
  margin {
    left: 0.4rem;
    bottom: 0.4rem
  }
}

@media (min-width: 1025px) and (max-width: 1280px) {
  .ui.form .fields .formGroup  {
    width: 50%
  }
}
  @media (max-width: 767px) {
  input#_password,input#password{
    width: 17.8rem !important;
  }
  }