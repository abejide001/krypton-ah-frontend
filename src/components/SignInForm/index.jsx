import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Form, Checkbox, Button } from 'semantic-ui-react';
import { Redirect } from 'react-router-dom';
import InlineError from '../../helpers/InlineError';
import { signInValidator } from '../../helpers/validateUser';
import { userLogin } from '../../actions/authAction/authActions';
import RedirectLoader from '../Loader';

class SignIn extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      user: {
        email: '',
        password: ''
      },
      errors: {}
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    event.preventDefault();
    const { user } = this.state;
    const errors = signInValidator(user);
    if (errors) {
      this.setState({ errors });
    }
    const { login } = this.props;
    login(user);
  }

  handleChange(event) {
    const { user } = this.state;
    const { name, value } = event.target;
    user[name] = value;
    this.setState({
      user
    });
  }

  render() {
    const {
      user, errors
    } = this.state;

    if (this.props.auth.authIsLoading) { return <RedirectLoader />; }
    if (this.props.auth.isAuthenticated) { return <Redirect to="/writearticle" />; }
    return (
      <Form onSubmit={this.handleSubmit}>
        <Form.Group widths="equal">
          <Form.Input
            type="email" className="email" fluid placeholder="Email Address" name="email" value={user.email} onChange={this.handleChange} required
          />
        </Form.Group>
        {errors.email && <InlineError text={errors.email} />}
        <Form.Group widths="equal">
          <Form.Input
            fluid type="password" placeholder="Password" value={user.password} name="password" onChange={this.handleChange} required
          />
        </Form.Group>
        { errors.password && <InlineError text={errors.password} />}
        <Form.Field>
          <Checkbox label="Remember me" />
        </Form.Field>
        <Button type="submit" basic fluid huge="true" onClick={this.handleSubmit}>Sign In</Button>
      </Form>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  login: user => dispatch(userLogin(user))
});

const mapStateToProps = state => ({
  auth: state.authReducer
});

export { SignIn as Login };
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SignIn);

SignIn.propTypes = {
  login: PropTypes.func,
  auth: PropTypes.object
};

SignIn.defaultProps = {
  login: null,
  auth: {}
};
