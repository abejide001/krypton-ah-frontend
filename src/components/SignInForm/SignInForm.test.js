import React from 'react';
import { shallow, mount } from 'enzyme';
import SignInForm from '.';

describe('SignIn Form', () => {
  it('should matches the snapshot', () => {
    const wrapper = shallow(<SignInForm />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should change the state after change the input value', () => {
    const newValue = 'iolan@anla.com';
    const wrapper = mount(<SignInForm />);
    wrapper.find('input[type="email"]').simulate('change', {
      target: { name: newValue }
    });
    wrapper.update();
    expect(wrapper.find('input[type="email"]').prop('name')).toContain('email');
    expect(wrapper.find('input[type="password"]').prop('name')).toContain('password');
  });

  it('should simulate on submit event ', () => {
    const wrapper = mount(<SignInForm />);
    wrapper.find('form').simulate('submit', {
      preventDefault: jest.fn()
    });
    expect(wrapper.find('form')).toBeDefined();
  });
});
