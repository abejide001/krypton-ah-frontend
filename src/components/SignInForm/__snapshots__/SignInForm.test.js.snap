// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Form should matches the snapshot 1`] = `
<Form
  as="form"
  onSubmit={[Function]}
>
  <FormGroup
    widths="equal"
  >
    <FormInput
      as={[Function]}
      className="email"
      control={[Function]}
      fluid={true}
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      required={true}
      type="email"
      value=""
    />
  </FormGroup>
  <FormGroup
    widths="equal"
  >
    <FormInput
      as={[Function]}
      control={[Function]}
      fluid={true}
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
  </FormGroup>
  <FormField>
    <Checkbox
      label="Remember me"
      type="checkbox"
    />
  </FormField>
  <Button
    as="button"
    basic={true}
    fluid={true}
    huge="true"
    onClick={[Function]}
    type="submit"
  >
    Sign In
  </Button>
</Form>
`;
